File: examples/twosum.php
<?php
function twoSum($nums, $target) {
    $map = [];  // associative array to store number => index

    foreach ($nums as $index => $num) {
        $complement = $target - $num;
        if (isset($map[$complement])) {
            return [$map[$complement], $index];
        }
        $map[$num] = $index;
    }

    return [];  // return empty array if no solution
}

// Example usage:
$numbers = [2, 7, 11, 15];
$target = 9;
$result = twoSum($numbers, $target);

print_r($result);  // Output: [0, 1]
?>

File: examples/twosum.py
def two_sum(nums, target):
    num_map = {}  # Store number as key and index as value
    for index, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], index]
        num_map[num] = index
    return []  # If no solution is found
